Converter/MapManager.cs

```diff
@@ -1181,4 +1198,198 @@ public static async Task WriteHolySites(Map map, string[] pickedFaiths)
+            if (baroniesInProvince.Count == 0)
+            {
+                Console.WriteLine($"No baronies in {province.Name}");
+                continue;
+            }
+
+            //Sort them on population size decending baroniesInProvince[0].Burg.population
+            baroniesInProvince.Sort((a, b) => b.Burg.Population.CompareTo(a.Burg.Population));
+
+            // Now get all cells in the province that are not already assigned to a burg (country side cells)
+            var CountrysideCells = province.Cells.Where(c => !c.hasBurg).ToDictionary(c => c.id, c => c);
+
+            // Have each barony grow outwards until all countryside cells are assigned among the burgs based on distance
+            while (CountrysideCells.Count > 0)
+            {
+                foreach (Barony barony in baroniesInProvince)
+                {
+                    //Look at it's assigned cells
+                    // And generate a list of neighbouring cells from the list of countryside cells
+                    var neighbors = barony.Cells.SelectMany(c => c.neighbors).Where(c => CountrysideCells.ContainsKey(c)).Select(c => CountrysideCells[c]).ToList();
+                    if (neighbors.Count == 0)
+                    {
+                        continue;
+                    }
+
+                    //Then sort them by distance to the burg
+                    neighbors.Sort((a, b) => a.DistanceSquared(barony.Burg.Cell).CompareTo(b.DistanceSquared(barony.Burg.Cell)));
+                    //Then assign the closest cell to the burg
+                    barony.Cells.Add(neighbors[0]);
+                    //And remove it from the list of countryside cells
+                    CountrysideCells.Remove(neighbors[0].id);
+                }
+            }
+
+        }
+    }
+
+    private static Dictionary<int, ConverterBurg> AddBurgCellReference(Dictionary<int, ConverterBurg> burgs, Dictionary<int, Cell> cells)
+    {
+        //For each burg (skip 0'eth) find the cell it is referenceing by cell id and assign it to the burg
+        foreach (var burg in burgs.Skip(1))
+        {
+            burg.Value.Cell = cells[burg.Value.azgaarBurg.cell];
+        }
+
+        return burgs;
+    }
+
+    /// <summary>
+    /// Generate a list of baronies with the bare minimum of information
+    /// </summary>
+    /// <param name="burgs">Burgs to base the baronies on</param>
+    /// <returns>A list of baronies</returns>
+    private static List<Barony> GenerateBaronies(Dictionary<int, ConverterBurg> burgs)
+    {
+        // Next we instanciate a list of baronies. Since we know the final size of the list we can set the capacity to the number of burgs
+        List<Barony> baronies = new(burgs.Count - 1);
+        foreach (var burg in burgs.Skip(1)) //0'eth entry is always empty (See Azgaar data model)
+        {
+            baronies.Add(new Barony(burg.Value));
```

Comment:
when instantiating with final size you need to set values by index instead of adding them.

Reply:
you don't really need a foreach here.
`return burgs.Skip(1).Select(n => new Barony(n.Value)).ToList();`

Reply:
The list does not have a final size, I just initialize  it with an appropriate size for optimisation reasons. That way it does not have to resize so often.

Reply:
Might do a vanity pass when I am done, but I find having the foreach be explisit makes it easier for me to read and modify while working on it.

Reply:
Hey, thank you for taking the time to review btw!